pkg/operation/controlplanesecrets/secrets_generator.go

```diff
@@ -0,0 +1,96 @@
+// Copyright (c) 2020 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package controlplanesecrets
+
+import (
+	gardencorev1alpha1helper "github.com/gardener/gardener/pkg/apis/core/v1alpha1/helper"
+	"github.com/gardener/gardener/pkg/utils/infodata"
+	"github.com/gardener/gardener/pkg/utils/secrets"
+)
+
+// SecretsGenerator extends the SecretsManager with methods which can generate InfoData from the given secret configurations.
+type SecretsGenerator struct {
```

Comment:
What's the purpose of this struct? It's only using the `SecretsManager` internally. Does it make sense to remove it and move the struct methods to the `SecretsManager`, instead?

Reply:
Wanted to keep the functions related to generating secrets separate from the SecretsDeployer struct, since it also uses thee SecretsManager struct.

Reply:
Why not combine everything together, see above.